Mutagen.Bethesda.Analyzers.Cli/Overrides/CustomEnabledPluginListingsProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Mutagen.Bethesda.Plugins;
+using Mutagen.Bethesda.Plugins.Order;
+using Mutagen.Bethesda.Plugins.Order.DI;
+namespace Mutagen.Bethesda.Analyzers.Cli.Overrides;
+
+public class CustomEnabledPluginListingsProvider(string path) : IEnabledPluginListingsProvider
```

Comment:
Hmm, so we're taking in a custom DataFolder, and using that to drive a replacement for Plugins.txt, just assuming any mod that is present is one to watch?

That's a bit deceiving as the CLI API suggests it's just setting the datafolder -> not that it's overriding the plugins.txt completely as well

What's the reasoning for wanting to override this in this manner?  For the continuous integration, i imagine?  If so, maybe we add another parameter `--UseDataFolderForLoadOrder` or something

------------

Alternatively, I think the analyzer project in general shouldn't be too concerned with plugins.txt overall.  Like, when a user opens up the analyzer UI, it should just analyze all the files in the data folder.  It should also consider plugins.txt for some specific analyzers, but that's more of an afterthought than a required input.

So, perhaps instead of adding `--UseDataFolderForLoadOrder` we add `--WithNoLoadOrder` and omit the plugins.txt concepts entirely (potentially turning off a few analyzers dedicated to analyzing load order interactions)

Reply:
Yeah I need it for CI. `--WithNoLoadOrder` wouldn't really work for my case when it would turn of the load order analyzers because I'd still wanted them in there, just in this different environment. Having an implicit `--UseDataFolderForLoadOrder` could be good I think.
Any analyzer UI would have their own registration so I think this would be entirely separate from the cli stuff.